hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsOutputStream.java

```diff
@@ -168,6 +177,10 @@ public synchronized void write(final byte[] data, final int off, final int lengt
       throw new IndexOutOfBoundsException();
     }
 
+    if (lease != null && lease.isFreed()) {
+      throw new IOException("Attempted to write to file without lease: " + path);
```

Comment:
PathIOException with path; make error string a const to use when matching in tests. Consider also a new LeaseRequiredException if that helps testing